src/main/java/com/mealmentor/enterprise/MealMentorController.java

```diff
@@ -77,37 +106,70 @@ public String read(Model model){
         return "start";
     }
 
-
+    /**
+     * Search recipe by the search term
+     * @param searchTerm the search term
+     * @param model model
+     * @return Return user back to the recipes page, with the loaded model object
+     * @throws IOException
+     */
     @GetMapping(value = "/searchRecipe")
     @ResponseBody
-    public String searchReceipe(@RequestParam(value="searchTerm", required=false, defaultValue="None")  String searchTerm, Model model) throws IOException {
+    public ModelAndView searchRecipe(@RequestParam(value="searchTerm", required=false, defaultValue="None")  String searchTerm, Model model) throws IOException {
+        ModelAndView modelAndView = new ModelAndView();
         try {
             List<Recipe> recipes= mealPlanService.fetchRecipes(searchTerm);
-            model.addAttribute("recipes", recipes);
-            return "recipes";
+            modelAndView.addObject("recipes", recipes);
+            modelAndView.setViewName("recipes");
+            return modelAndView;
+
         } catch (IOException e) {
-            e.printStackTrace();
-            return "error";
+            logger.error("problem in searchRecipe.");
+            modelAndView = createErrorModelAndView("There was a problem with the search",
+                    "please restart the application and let the admins know if the problem persists");
+            modelAndView.setViewName("error");
+            return modelAndView;
         }
 
```

Comment:
I changed the method type to **ModelAndView**. This is so that in case of an error, you can send additional information to the error page, instead of displaying a generic error message. Two of ModelAndView's most common methods are **addObject** and **setViewName**, examples of which you can see in the method.